### PR TITLE
Feature/delete event++

### PR DESCRIPTION
**Pull request by FredrikMorstad**
## Delete events via admin->event table
 - Dependant on https://github.com/td-org-uit-no/tdctl-api/pull/63
 
Screenshot from 2023-02-17 15-45-39
### Minor changes:
  - Close modal on `<esc>`
  - Update message for users leaving a event within 24 hours 
  
image

### CODE REVIEW

File: src/components/molecules/event/eventButton/EventButton.tsx [hunk 1]
```diff
@@ -177,9 +177,17 @@ const AuthEventButton: React.FC<AuthButtonProps> = ({
...
+                  style={{
+                    textDecorationColor: '#b73653',
+                    textDecorationThickness: '.2rem',
+                  }}>
+                  {' '}
+                  en merknad hvis du har mottat bekreftelse om plass
+                </u>
+                . To eller flere merknader vil gi nedsatt prioritet på andre
+                arrangementer ut semesteret. Har du gyldig grunn ta kontakt med{' '}
                 <a href="mailto:">{event?.host ?? 'post.td.uit.no'}</a>
```
Comment (seblj) [inline src/components/molecules/event/eventButton/EventButton.tsx:191]: I think it's the wrong email here. Try to grep for td.uit.no and see if there is other places we need to change

File: src/components/molecules/tables/eventTable/eventTable.tsx [hunk 2]
```diff
@@ -5,15 +5,81 @@ import { useHistory } from 'react-router-dom';
...
+    // display error toast on error caught by useUpcomingEvents hook
+    if (error !== null) {
+      addToast({
+        title: 'Kunne ikke laste inn arrangementetene',
+        status: 'error',
+      });
+    }
+  }, [error]);
+
+  const { confirmed, setConfirmed } = useConfirmation(adminDeleteEvent);
```
Comment (seblj) [inline src/components/molecules/tables/eventTable/eventTable.tsx:81]: Why do we need this? Can't we just call `adminDeleteEvent` in the `onAccept` prop in the confirmation component?

`onDecline` could be set to an optional prop in the component so we don't do anything in that case

Then we can also remove the check for confirmation in the delete function

File: src/components/molecules/tables/eventTable/eventTable.tsx [hunk 3]
```diff
@@ -65,12 +131,44 @@ const EventTable = () => {
...
+    },
   ];
 
   return (
     <>
       <div className={styles.form}>
         <Table columns={columns} data={events}></Table>
+        {deleteModal && (
+          <Modal
+            title={`Are you sure you want to delete ${selectedEvent?.title}?`}
```
Comment (seblj) [inline src/components/molecules/tables/eventTable/eventTable.tsx:159]: Norwegian?

File: src/components/molecules/tables/eventTable/eventTable.tsx [hunk 4]
```diff
@@ -5,15 +5,81 @@ import { useHistory } from 'react-router-dom';
...
+  const openDeleteColumn = (eid: string) => {
+    const selected = events?.find((mem) => {
+      return mem.eid === eid;
+    });
+    setSelected(selected);
+    setOpenDeleteModal(true);
+  };
+
+  const adminDeleteEvent = async () => {
+    setOpenDeleteModal(true);
```
Comment (seblj) [inline src/components/molecules/tables/eventTable/eventTable.tsx:37]: Why is this needed? Isn't this already open when this is called?

File: src/components/molecules/tables/eventTable/eventTable.tsx [hunk 5]
```diff
@@ -65,12 +131,44 @@ const EventTable = () => {
...
       <div className={styles.form}>
         <Table columns={columns} data={events}></Table>
+        {deleteModal && (
+          <Modal
+            title={`Are you sure you want to delete ${selectedEvent?.title}?`}
+            setIsOpen={setOpenDeleteModal}
+            minWidth={45}>
+            <ConformationBox
+              onAccept={() => {
+                setConfirmed(true);
```
Comment (seblj) [inline src/components/molecules/tables/eventTable/eventTable.tsx:164]: `adminDeleteEvent` should just be called here

File: src/components/molecules/tables/eventTable/eventTable.tsx [hunk 6]
```diff
@@ -65,12 +131,44 @@ const EventTable = () => {
...
+              }}
+              onDecline={() => {
```
Comment (seblj) [inline src/components/molecules/tables/eventTable/eventTable.tsx:166]: This could be made optional and be removed here when the other comments are adressed

File: src/hooks/useEvents.ts [hunk 7]
```diff
@@ -14,25 +16,33 @@ const useUpcomingEvents = () => {
...
-        setError(error.statusCode);
-      } finally {
-        setIsFetching(false);
-      }
-    };
     fetchEvents();
   }, [authenticated]);
 
-  return { isFetching, events, setEvents, error };
+  useEffect(() => {
```
Comment (seblj) [inline src/hooks/useEvents.ts:38]: Can't we just return `fetchEvents` from this hook and get rid og this useEffect?.

Then we can just call `fetchEvents()` directly instead of setting a state to trur which triggers a useEffect that fetched events and immediately sets the state to false again

**Review by seblj — CHANGES_REQUESTED** [review]
I think there is some places we can remove some code while still having the same functionality

File: src/hooks/useEvents.ts (hunk 7 above)
Comment (FredrikMorstad) [reply src/hooks/useEvents.ts:38]: The useEffect with regard to the authentication parameter is there to ensure when authentication change on the client, the correct data is collected from the API. For instance, when you are automatically logged out as admin you want to show events that should only be visible for non-auth users

Comment (FredrikMorstad) [reply src/hooks/useEvents.ts:38]: Ahhh I see, yes that could be done :+1:

File: src/components/molecules/tables/eventTable/eventTable.tsx (hunk 3 above)
Comment (FredrikMorstad) [reply src/components/molecules/tables/eventTable/eventTable.tsx:159]: oops, yes that should be Norwegian

File: src/components/molecules/tables/eventTable/eventTable.tsx (hunk 4 above)
Comment (FredrikMorstad) [reply src/components/molecules/tables/eventTable/eventTable.tsx:37]: yes, kinda defeated the point of having it in a hook

File: src/components/molecules/event/eventButton/EventButton.tsx (hunk 1 above)
Comment (FredrikMorstad) [reply src/components/molecules/event/eventButton/EventButton.tsx:191]: whoops :pencil2:

File: src/components/molecules/tables/eventTable/eventTable.tsx (hunk 2 above)
Comment (FredrikMorstad) [reply src/components/molecules/tables/eventTable/eventTable.tsx:81]: Yes that would be smoother, changing now

File: src/components/molecules/tables/eventTable/eventTable.tsx (hunk 6 above)
Comment (FredrikMorstad) [reply src/components/molecules/tables/eventTable/eventTable.tsx:166]: I still thing onDecline needs to be required as the modal needs to be closed onDecline

Comment (seblj) [reply src/components/molecules/tables/eventTable/eventTable.tsx:166]: Yes of course that makes sense🙌

Comment (FredrikMorstad) [reply src/components/molecules/tables/eventTable/eventTable.tsx:166]: Almost worked, however the modal setter can not set the confirmed state back to undefined as it only set boolean values. I think adding a onClick prop to only solve this case is not suitable

File: src/components/molecules/tables/memberTable/MemberTable.tsx [hunk 8]
```diff
@@ -13,24 +13,28 @@ import { RoleOptions } from 'contexts/authProvider';
...
-  const [isOpen, setIsOpen] = useState(false);
-  const [deleteModal, setOpenDeleteModal] = useState(false);
   const [selectedMember, setSelected] = useState<Member | undefined>();
   const [initialValues, setInitialValues] = useState<AdminMemberUpdate>();
   const [error, setError] = useState<undefined | string>();
 
   const { addToast } = useToast();
+  const { isOpen, onOpen, onClose } = useModal();
+  const {
+    isOpen: IsOpenDelteModal,
```
Comment (seblj) [inline src/components/molecules/tables/memberTable/MemberTable.tsx:34]: camelCase and `Delete` instead of `Delte`

File: src/components/molecules/event/eventButton/EventButton.tsx [hunk 9]
```diff
@@ -35,37 +36,39 @@ const AuthEventButton: React.FC<AuthButtonProps> = ({
...
   const [showAllergies, setShowAllergies] = useState<boolean>(false);
   const [joinEventPayload, setJoinEventPayload] = useState<JoinEventPayload>({
     food: false,
     transportation: false,
     dietaryRestrictions: '',
   });
+  const { addToast } = useToast();
+  const { isOpen, onOpen, onClose } = useModal();
+  const {
+    isOpen: preferancesOpen,
```
Comment (seblj) [inline src/components/molecules/event/eventButton/EventButton.tsx:50]: `preferances` -> `preferences` for all of these

File: src/components/molecules/event/eventResponses/EventResponses.tsx [hunk 10]
```diff
@@ -169,45 +164,44 @@ const EventResponses: React.FC<{ event: Event; setFetchUpdateHook: React.Dispatc
...
+          </div>
         </form>
-      )}
+      </Modal>
+      <Modal
+        title={`Remove ${selectedParticipant?.realName ?? ''}?`}
+        isOpen={isOpenDeleteModal}
+        onClose={closeDeleteModal}
+        minWidth={45}>
+        <ConformationBox
```
Comment (seblj) [inline src/components/molecules/event/eventResponses/EventResponses.tsx:201]: `ConformationBox` -> `ConfirmationBox`

**Review by seblj — APPROVED** [review]
Awesome! Looks really good 🙌 Just some small typos that can be changed if you bother, but otherwise I approve
